compositor_pipeline/src/pipeline/input/rtp/depayloader.rs

```diff
@@ -142,21 +164,103 @@ impl AudioDepayloader {
         packet: rtp::packet::Packet,
     ) -> Result<Option<EncodedChunk>, DepayloadingError> {
         match self {
-            AudioDepayloader::Opus(depayloader) => {
+            AudioDepayloader::Opus {
+                depayloader,
+                previous_timestamp,
+                rollover_count,
+            } => {
                 let kind = EncodedChunkKind::Audio(AudioCodec::Opus);
                 let opus_packet = depayloader.depacketize(&packet.payload)?;
 
                 if opus_packet.is_empty() {
                     return Ok(None);
                 }
 
+                let timestamp = timestamp_with_rollover(
+                    rollover_count,
+                    *previous_timestamp,
+                    packet.header.timestamp,
+                );
+
+                *previous_timestamp = packet.header.timestamp;
+
                 Ok(Some(EncodedChunk {
                     data: opus_packet,
-                    pts: Duration::from_secs_f64(packet.header.timestamp as f64 / 48000.0),
+                    pts: Duration::from_secs_f64(timestamp as f64 / 48000.0),
                     dts: None,
                     kind,
                 }))
             }
         }
     }
 }
+
+fn timestamp_with_rollover(
+    rollover_count: &mut usize,
+    previous_timestamp: u32,
+    current_timestamp: u32,
+) -> u64 {
+    let timestamp_diff = u32::abs_diff(previous_timestamp, current_timestamp);
+    if timestamp_diff >= u32::MAX / 2 {
+        if previous_timestamp > current_timestamp {
+            *rollover_count += 1;
+        } else {
+            // We received a packet from before the rollover, so we need to decrement the count
+            *rollover_count = rollover_count.saturating_sub(1);
```

Comment:
If first timestamp is larger than `u32:MAX/2` then this will try to decrement 0, which will result in a zero value. This behavior is correct, but I'm not a fan how this hides the case.

Either:
- describe this case in a comment
- or use Option<u32> for previous timestamp